File: app/channels/chatroom_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ChatroomChannel", {
  connected() {
    console.log("Connected to the chatroom!")
  },

  disconnected() {
  },

  received(data) {
    const messages = document.getElementById('messages')
    messages.insertAdjacentHTML('beforeend', data['message'])
  }
})

File: app/channels/chatroom_channel.rb
class ChatroomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chatroom_channel"
  end

  def unsubscribed
  end
  received(data) {
    const messages = document.getElementById('messages')
    messages.insertAdjacentHTML('beforeend', data['message'])
  }
end
